### PR TITLE
h4

### PR DESCRIPTION
**Pull request by dr-programmer**

### CODE REVIEW

**Comment by dr-programmer** [comment]
@VladislavVulchevChaos

File: SourceCode/h4/triangle1.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        this->y = y;
+        this->z = z;
+    }
+
+    float length() const {
+        return sqrt(x * x + y * y + z * z);
+    }
+    void normalize() {
+        float cachedLength = length();
+        x = x / cachedLength;
```
Comment (VladislavVulchevChaos) [inline SourceCode/h4/triangle1.cpp:18]: You could use only 1 division, just in case :)

File: SourceCode/h4/triangle1.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        y = y / cachedLength;
+        z = z / cachedLength;
+    }
+
+    float getX() const {
```
Comment (VladislavVulchevChaos) [inline SourceCode/h4/triangle1.cpp:23]: Since you have getters and setters for all the members you could use a struct directly.

Comment (dr-programmer) [reply SourceCode/h4/triangle1.cpp:23]: That's a good point. Will add.

File: SourceCode/h4/triangle1.cpp (hunk 1 above)
Comment (dr-programmer) [reply SourceCode/h4/triangle1.cpp:18]: What do you mean 1 division? I have to apply it to all three coordinates, or am I not seeing something?

**Pull request merged**